코드리뷰용 PR

clara-shin opened the pull request:
# 🚀 풀 리퀘스트 제안

<!-- 이슈 지울때 작성해 주세요. -->
<!-- closes #issue-number -->
<!-- 이슈 여러 개 일 경우, closes #1, closes #2 -->
<!-- https://docs.github.com/ko/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue -->

## 📋 작업 내용

수정한 내용이나 추가한 기능에 대해 자세히 설명해 주세요.

## 🔧 변경 사항

- [ ] 📃 README.md
- [ ] 📦 package.json
- [ ] 🔥 파일 삭제
- [ ] 🧹 그 외 ex) .gitignore 등

주요 변경 사항을 요약해 주세요.

## 📸 스크린샷 (선택 사항)

수정된 화면 또는 기능을 시연할 수 있는 스크린샷을 첨부해 주세요.

## 📄 기타

추가적으로 전달하고 싶은 내용이나 특별한 요구 사항이 있으면 작성해 주세요.

<!-- Generated by sourcery-ai[bot]: start summary -->

## Sourcery에 의한 요약

개인화된 재생 목록, 사용자 프로필, Firebase 통합과 같은 기능을 갖춘 비디오 재생 목록 공유 웹 앱을 개발합니다. 리팩토링된 구조, 개선된 ESLint 구성, 포괄적인 문서를 통해 프로젝트를 향상시킵니다. 반응형 디자인을 구현하고 Playwright를 도입하여 종단 간 테스트를 수행합니다.

새로운 기능:
- 사용자가 개인화된 재생 목록을 생성, 공유 및 탐색할 수 있는 포괄적인 비디오 재생 목록 공유 웹 앱 서비스를 도입합니다. 비디오 추천, 구독 타임라인, 미니 비디오 플레이어와 같은 기능을 제공합니다.
- 역할과 책임이 있는 사용자 프로필 시스템을 구현하여 사용자가 앱 내에서 재생 목록, 댓글 및 상호작용을 관리할 수 있도록 합니다.
- 데이터베이스 관리, 인증 및 호스팅을 위한 Firebase 통합 지원을 추가하여 앱의 확장성과 성능을 향상시킵니다.
- 재생 목록 내 비디오 순서를 관리하기 위한 드래그 앤 드롭 인터페이스를 도입하여 보다 직관적인 사용자 경험을 제공합니다.
- 효율적인 클라이언트 및 서버 상태 관리를 위해 Zustand 및 TanStack Query를 사용하여 글로벌 상태 관리 시스템을 구현합니다.
- 사용자가 카테고리 및 키워드로 재생 목록을 검색할 수 있는 포괄적인 검색 및 필터 기능을 추가합니다.
- 모바일 우선 접근 방식을 통해 반응형 디자인을 도입하여 기기 전반에 걸쳐 최적의 사용자 경험을 보장합니다.

향상된 기능:
- 구성 요소, 훅 및 API 엔드포인트의 분리를 포함하여 코드 조직 및 유지 관리를 개선하기 위해 프로젝트 구조를 리팩토링합니다.
- 프로젝트 전반에 걸쳐 코딩 표준을 적용하고 코드 품질을 향상시키기 위해 ESLint 구성을 강화합니다.
- 프로젝트에 대한 설정 지침, 팀 역할 및 프로젝트 목표를 포함한 자세한 정보를 제공하기 위해 README.md를 업데이트합니다.
- 사용자 상호작용 및 피드백을 개선하기 위해 사용자 정의 대화 상자 및 토스트 알림 시스템을 구현합니다.
- CSS-in-JS 스타일링을 위한 Emotion을 사용하여 UI를 향상시켜 동적이고 유지 관리 가능한 스타일을 제공합니다.

빌드:
- 개발 워크플로우 및 빌드 성능을 개선하기 위해 TypeScript 경로 및 스타일링을 위한 Emotion 지원을 포함하도록 Vite 구성을 업데이트합니다.

CI:
- 애플리케이션의 신뢰성과 안정성을 보장하기 위해 종단 간 테스트를 위한 Playwright 구성을 추가합니다.

문서화:
- 개발자 온보딩 및 협업을 돕기 위해 프로젝트의 API 엔드포인트, 구성 요소 및 훅에 대한 자세한 문서를 추가합니다.

테스트:
- 애플리케이션 내의 중요한 사용자 흐름 및 상호작용을 다루는 종단 간 테스트를 위해 Playwright 테스트를 도입합니다.

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Develop a video playlist sharing web app with features like personalized playlists, user profiles, and Firebase integration. Enhance the project with a refactored structure, improved ESLint configuration, and comprehensive documentation. Implement a responsive design and introduce Playwright for end-to-end testing.

New Features:
- Introduce a comprehensive video playlist sharing web app service, allowing users to create, share, and explore personalized playlists with features like video recommendations, subscription timelines, and a mini video player.
- Implement a user profile system with roles and responsibilities, enabling users to manage their playlists, comments, and interactions within the app.
- Add support for Firebase integration for database management, authentication, and hosting, enhancing the app's scalability and performance.
- Introduce a drag-and-drop interface for managing video order within playlists, providing a more intuitive user experience.
- Implement a global state management system using Zustand and TanStack Query for efficient client and server state management.
- Add a comprehensive search and filter functionality, allowing users to search playlists by categories and keywords.
- Introduce a responsive design with a mobile-first approach, ensuring optimal user experience across devices.

Enhancements:
- Refactor the project structure to improve code organization and maintainability, including the separation of components, hooks, and API endpoints.
- Enhance the ESLint configuration to enforce coding standards and improve code quality across the project.
- Update the README.md to provide detailed information about the project, including setup instructions, team roles, and project goals.
- Implement a custom dialog and toast notification system for better user interaction and feedback.
- Enhance the UI with Emotion for CSS-in-JS styling, providing dynamic and maintainable styles.

Build:
- Update the Vite configuration to include support for TypeScript paths and Emotion for styling, improving the development workflow and build performance.

CI:
- Add Playwright configuration for end-to-end testing, ensuring the reliability and stability of the application.

Documentation:
- Add detailed documentation for the project's API endpoints, components, and hooks, aiding in developer onboarding and collaboration.

Tests:
- Introduce Playwright tests for end-to-end testing, covering critical user flows and interactions within the application.

</details>

<!-- Generated by sourcery-ai[bot]: end summary -->

miniseung committed:
feat: 플레이리스트 수정 (#126)

-커스텀훅 되돌리기
-수정api 작성
-PlaylistEidt페이지 추가

dyeongg committed:
feat: 온보딩페이지와 로그인페이지에서 로그인후 홈까지 오는 테스트 코드 작성 (#127)

- 온보딩페이지에서 로그인페이지로 넘어가는 테스트
- 로그인페이지에서 값 입력하고 홈으로 가는 테스트

clara-shin committed:
feat: VideoModal 전역으로 관리

- VideoModal을 전역으로 관리하기 위해 useVideoModalStore.ts 파일 생성
- VideoModal을 열고 닫는 함수를 store에 추가
- RootLayout에서 VideoModal을 사용할 수 있도록 수정
- PlaylistPage에서 openMiniPlayer 함수를 store에서 가져와 사용하도록 수정

miniseung committed:
chore: 공개 비공개 문구 출력 (#126)

unanbb committed:
Merge pull request #131 from Dev-FE-1/feat/commentPage-#115

[Feat] 댓글 목록 페이지 퍼블리싱 및 데이터베이스 연동

dyeongg committed:
Merge pull request #134 from Dev-FE-1/refactor/login-test-#127

[Refactor] playwright로 온보딩페이지와 로그인페이지에서 로그인후 홈까지 오는 테스트 코드 작성

clara-shin committed:
feat: 전역에서 플레이리스트의 영상을 변경할 수 있도록 구현 (#116)

- useMiniPlayerStore에 updatePlaylist 함수 추가: 현재 재생중인 영상의 플레이리스트 정보를 함께 업데이트
- PlaylistPage에서 handleVideoClick 함수 수정를 사용하여 플레이리스트 정보도 함께 전달
- VideoModal에서 PlaylistPage에서 전달받은 플레이리스트 정보를 사용하여 영상 변경
    - 이 때, 영상 변경 시, 플레이리스트 정보도 함께 업데이트

clara-shin committed:
fix: 미니플레이어가 플리 삭제모달보다 위에 뜨는 문제 수정 (#116)

- 미니플레이어가 재생 중인 상태에서 플리 전체삭제 모달이 미니플레이어보다 더 위에 뜨도록 수정

miniseung committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into feat/playlist-api-#126

miniseung committed:
Merge pull request #136 from Dev-FE-1/feat/playlist-api-#126

[Feat] 플레이리스트 메인페이지 수정,삭제,비디오삭제

clara-shin committed:
feat: 미니 플레이어가 떠 있어도 그 밑의 컴포넌트도 이동 가능하도록 스크롤바 활성화

- body에 스크롤바가 인식되지 않도록 했던 코드를 삭제

clara-shin committed:
fix: PlaylistPage에 미니 플레이어가 떠 있을 때 PlaylistPage의 헤더를 클릭하면 음악재생순서의 뒤로 이동하는 문제 수정 (#116)

- PlaylistPage에서 특정 영상을 클릭해서 VideoModal을 띄운 후, 작은 미니 플레이어가 떠 있을 때
- PlaylistPage의 헤더를 클릭하면 음악재생순서의 뒤로 이동하는 문제를 수정하고, 마이페이지로 이동하도록 경로 수정

clara-shin committed:
feat: 미니플레이어 재생중에 로그아웃 하면 미니플레이어 초기화 (#116)

- 미니플레이어가 초기화되며 사라지고 로그아웃되어 로그인 페이지로 이동

clara-shin committed:
Merge branch 'develop' into feat/playlist-detail-#116

unanbb committed:
chore: useComments.ts 불필요한 코드 삭제 (#115)

userComments에 사용하지 않는 import 코드 삭제

dyeongg committed:
feat: 댓글 작성 페이지 퍼블리싱 완료 (#135)

댓글 작성시 보여지는 페이지 퍼블리싱 완료

itsnowkwon committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into feat/video-create-#124

itsnowkwon committed:
fix: 플레이리스트 수정 시 카테고리 나오지 않는 버그 수정 (#124)

- 기존 `option.label`를 `option.value`로 수정

i

unanbb committed:
feat: 로그인한 사용자의 댓글에 삭제버튼 추가 (#115)

- 로그인한 사용자의 댓글에만 삭제버튼이 보이도록 구현
- Comment 타입에 사용하지 않는 updatedAt 변수 삭제

miniseung committed:
feat: 마이페이지 유저별 랜더링 (#137)

unanbb committed:
chore: 댓글 삭제 버튼 크기 조정 (#115)

- 사용성을 위해 댓글 삭제버튼 크기 16x16 -> 20x20로 변경

unanbb committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into feat/commentPage-#115

miniseung committed:
chore: 오타수정 (#137)

miniseung committed:
chore: 불필요한선언제거 (#137)

itsnowkwon committed:
Merge pull request #140 from Dev-FE-1/feat/video-create-#124

[Feat] 플레이리스트 메인 페이지에서 유튜브 API를 사용해 동영상 등록

miniseung committed:
chore: 헤더위치수정 (#137)

dyeongg committed:
Merge pull request #139 from Dev-FE-1/feat/comment-create-publishing-#135

[feat] 댓글 작성 페이지 퍼블리싱

itsnowkwon committed:
Merge branch 'feat/playlist-detail-#116' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into feat/playlist-detail-#116

itsnowkwon committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into feat/playlist-detail-#116

itsnowkwon committed:
Merge pull request #138 from Dev-FE-1/feat/playlist-detail-#116

[Feat] 상세 비디오모달 컴포넌트 구현(전역 미니 플레이리스트 함께 구현)

itsnowkwon committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into feat/toggle-button-api-#128

miniseung committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into feat/mypage-#137

miniseung committed:
Merge pull request #141 from Dev-FE-1/feat/mypage-#137

[Feat] 마이페이지 유저별 랜더링

itsnowkwon committed:
rename: 'ThumBox' 파일명을 'ThumbNailBox`로 가독성 있게 수정 (#128)
- 추가로 `nullBox.tsx`로 컨벤션에 맞게 `NullBox.tsx`로 수정

itsnowkwon committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into feat/toggle-button-api-#128

dyeongg committed:
feat: 댓글 작성 기능 구현 완료 (#142)

- 댓글 작성 후 파이어베이스에 저장
- 댓글 작성,실패시 토스트 알림
- 댓글 작성하면 바로 리스트에 반영됨

unanbb committed:
feat: 내 댓글 삭제 기능 구현 (#115)

- 내 댓글 삭제하기 (서버, 클라이언트)
- Comment 타입의 commnetId 타입 수정

itsnowkwon committed:
fix: TanStack Query의 `undefined` 반환으로 생기는 버그 수정 (#143)

- TanStack Query과 같은 데이터 패칭 라이브러리의 경우 초기에 `undefined`를 반환
- 이로 인해 생기는 문제를 예외처리로 해결

clara-shin committed:
feat: 플레이리스트의 비디오 순서 변경 후 업데이트해주는 함수 추가 (#117)

- 플레이리스트의 비디오 순서를 변경하면 변경된 순서대로 업데이트해주는 함수

clara-shin committed:
feat: 필요한 라이브러리 및 Strict Mode 해지

- react-beautiful-dnd 라이브러리 추가
- 해당 라이브러리 사용 시, React18에서는 strictMode를 꺼서 사용하라고 권장
    - StrictMode에서는 일부 라이브러리가 제대로 작동하지 않는 이슈가 있음
    - 해당 이슈로 인해 StrictMode를 해제하고 사용하기로 결정
- 대체 라이브러리도 있지만 react-beautiful-dnd 사용 결정
- 대체 라이브러리: @hello-pangea/dnd (react-beautiful-dnd의 포크 버전으로, 최신 React 버전과의 호환성 문제를 해결함)
```bash
npm uninstall react-beautiful-dnd
npm install @hello-pangea/dnd
```

clara-shin committed:
Merge pull request #144 from Dev-FE-1/feat/comment-add-#142

[feat] 플레이리스트 댓글 작성 기능

clara-shin committed:
Merge pull request #145 from Dev-FE-1/feat/commentPage-#115

[Feat] 내 댓글 삭제 기능 구현

clara-shin committed:
Merge pull request #146 from Dev-FE-1/develop

develop 브랜치에서 main 브랜치로 1차 병합

clara-shin committed:
feat: PlaylistPage와 VideoModal에 재생목록 드래그 앤 드롭 기능 추가 (#117)

- 위아래로 재생목록을 드래그하여 순서를 변경할 수 있도록 구현
- 재생목록의 순서가 변경되면 VideoModal에서도 순서가 변경되도록 구현
- 재생목록의 순서가 변경되면 PlaylistPage에서도 순서가 변경되도록 구현

dyeongg committed:
feat: 댓글 카운트, 댓글 작성시 카운트 +1

- 플레이리스트를 가져올때 댓글수 가져오기
- 댓글 작성시 기존 댓글에서 카운트 +1

clara-shin committed:
Merge pull request #149 from Dev-FE-1/feat/playlist-detail-#116

[Feat] 플레이리스트 상세페이지 Draggable 기능

clara-shin committed:
rename: hook폴더의 query폴더명 및 그 안의 파일명 변경

- query폴더명을 queries로 변경
- query폴더 안의 파일명을 useComments.ts -> useCommentsQueries.ts로 변경
- query폴더 안의 파일명을 usePlaylist.ts -> usePlaylistQueries.ts로 변경
- query폴더 안의 파일명을 useYoutube.ts -> useYoutubeQueries.ts로 변경
- hooks/mutations 폴더 추가

itsnowkwon committed:
fix: 메인페이지, 검색페이지에서 플레이리스트 항목 클릭 시 해당 플레이리스트 페이지로 이동 (#143)
- 기존에는 클릭해도 이동하지 않는 버그가 있었음

unanbb committed:
design: 댓글 이미지, max-width 수정 (#115)

- 댓글 이미지 object-fit: cover 추가
- 날짜 태그 max-width: 100px로 변경

itsnowkwon committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into fix/main-navigate-playlistMain-#143

unanbb committed:
Merge pull request #152 from Dev-FE-1/feat/commentPage-#115

[Design] 댓글 css 일부 수정

itsnowkwon committed:
Merge pull request #150 from Dev-FE-1/feat/commentadd-count-#147

[feat] 기존 댓글 카운트, 댓글 작성시 카운트 +1

itsnowkwon committed:
Revert "[feat] 기존 댓글 카운트, 댓글 작성시 카운트 +1"

itsnowkwon committed:
fix: 검색페이지 카테고리 필터링이 되지 않는 버그 수정 (#143)

- 여행 등록 시 카테고리를 영어로 저장하는 로직 삭제

dyeongg committed:
fix: 모달 컴포넌트, 첫진입시 404 에러 해결

- 머지하면서 사라진 모달 컴포넌트 내용 추가
- 첫 진입시 네비바 컴포넌트를 불러오며 에러가떠서 세션정보가 없으면 불러오지 않도록 코드 수정

itsnowkwon committed:
Merge pull request #154 from Dev-FE-1/revert-150-feat/commentadd-count-#147

Revert "[feat] 기존 댓글 카운트, 댓글 작성시 카운트 +1"

dyeongg committed:
Merge remote-tracking branch 'origin/develop' into feat/commentadd-count-#147

itsnowkwon committed:
feat: 좋아요 토글 버튼 클릭 시, 서버에 결과 반영 (#128)

- 사용자의 해당 플레이리스트 좋아요 여부 체크
- 좋아요가 되어있다면, 클릭했을 때 좋아요 +1
- 반대일 경우, 좋아요 -1

itsnowkwon committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into feat/toggle-button-api-#128

itsnowkwon committed:
comment: 파이어베이스 함수의 역할을 알기 쉽게 주석 추가 (#128)

itsnowkwon committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into fix/main-navigate-playlistMain-#143

itsnowkwon committed:
Merge pull request #153 from Dev-FE-1/fix/main-navigate-playlistMain-#143

[Fix] 메인페이지 플레이리스트 항목 클릭 시, 해당 플레이리스트 페이지로 이동 안 되는 버그 수정

dyeongg committed:
fix: revert한 댓글 카운트 기능 추가

댓글 카운트 기능을 추가합니다

dyeongg committed:
Merge pull request #155 from Dev-FE-1/feat/commentadd-count-#147

[fix] 모달 컴포넌트, 첫진입시 404 에러 해결

itsnowkwon committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into feat/toggle-button-api-#128

itsnowkwon committed:
Merge pull request #156 from Dev-FE-1/feat/toggle-button-api-#128

[Feat] 토글(좋아요, 구독) 버튼 클릭 시, 서버 데이터에 업데이트 요청 기능 구현

itsnowkwon committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into feat/fork-toggle-button-#132

itsnowkwon committed:
feat: 구독 토글 버튼 클릭 기능 구현 (#132)

- 구독 여부에 따른 아이콘 동적 렌더링
- 구독 중인 경우, 클릭하면 DB의 `userPlaylists` 컬렉션의 `forked` 필드에서 해당 플레이리스트 삭제
- 반대인 경우, `forked` 필드에 추가

clara-shin committed:
Merge branch 'develop' into feat/newPlaylist-refetch-#151

clara-shin committed:
feat: usePlaylistStore 수정

- addPlaylist -> addNewPlaylist 함수명 변경 내부 코드 수정
- zustand와 TanStack 사용하여 playlist store 구현 예정

clara-shin committed:
fix: 컴포넌트 명 및 파일경로 수정

Thumbox => ThumbNailBox 로 변경된 것에 대한 import 경로 수정

clara-shin committed:
fix: 썸네일 이미지가 없다면 디폴트 이미지가 보이도록 처리

clara-shin committed:
feat: 새 재생목록 생성 요청 중 버튼에 보여질 스피너 추가

clara-shin committed:
chore: 불필요한 주석삭제

clara-shin committed:
feat: 리액트 쿼리키 추가 및 유저 쿼리 추가

- 세션으로부터 받은 uid를 이용해 유저 정보를 가져오는 쿼리 추가
- useUserQueries.ts 파일 추가

clara-shin committed:
feat: 리액트쿼리를 사용하여 서버상태를 관리하는 로직 추가

- 기존의 useState와 useEffect를 걷어내고, 리액트쿼리를 사용하여 서버요쳥하는 api함수만 받아와서 구현
- queries
    - 마이페이지 컴포넌트가 마운트 되었을 떄 로그인한 사용자의 uid를 서버로 전송하여 사용자 정보를 받아오는 기능
    - 마이페이지 컴포넌트가 마운트 되었을 떄 로그인한 사용자의 uid를 서버로 전송하여 사용자의 플레이리스트 정보를 받아오는 기능
- mutations
    - 새 플레이리스트 추가 기능
- 추가적으로 새 플레이리스트 생성 중 마이페이지로 가기 전에 버튼 안에 스피너 추가

clara-shin committed:
Merge pull request #159 from Dev-FE-1/feat/fork-toggle-button-#132

[Feat] 구독 토글 버튼에 API 연결해 클릭마다 요청 보내는 기능 구현

itsnowkwon committed:
fix: TanStack Query에서 발생할 수 있는 초기 `undefined` 반환 버그 수정 (#132)

dyeongg committed:
feat: 유튜브 공유하기Url로 영상 추가

유튜브 공유하기 버튼을 통해 전달받은 url로 영상 추가하는 로직 추가

dyeongg committed:
feat: 유튜브 쇼츠 Url 처리

- 유튜브 쇼츠 영상 Url 처리
- switch 문으로 변경

dyeongg committed:
feat: 탐색페이지 카테고리 가로 스크롤 구현

탐색 페이지에 있는 카테고리 버튼들을 가로스크롤 가능하도록 변경

dyeongg committed:
refactor: 가로스크롤을  커스텀 훅으로 추출

- 가로스크롤 커스텀 훅 작성
- 홈페이지 가로스크롤 커스텀훅으로 변경
- 탐색페이지 카테고리 가로스크롤을 커스텀훅으로 변경

dyeongg committed:
fix: 미사용한 코드 제거 (#163)

clara-shin committed:
Merge pull request #161 from Dev-FE-1/feat/newPlaylist-refetch-#151

[Feat] 새 재생목록 생성 후 마이페이지로 이동 시 refetch하여 저장된 데이터까지 한번에 렌더링 되는 기능+ 리액트쿼리로 코드 변환

clara-shin committed:
Merge pull request #162 from Dev-FE-1/feat/youtube-link-#157

[Feat] 유튜브 공유하기링크, 쇼츠 영상 링크 처리

dyeongg committed:
feat: 홈에서 최신 플리 무한스크롤 5개씩 데이터 불러오기

홈에 있는 최신업데이트 플리에서 무한스크롤로 5개씩 데이터를 받아옵니다

dyeongg committed:
feat: 데이터가 없을 경우 무한 호출 방지

- 데이터가 더이상 없을 경우 무한 호출되지 않도록 코드 수정
- 처음 로드시 5개만 불러오고 그 이후부터 +5

clara-shin committed:
fix: 비디오모달에서 비디오목록에 스크롤이 생기지 않는 문제 수정

- controlsContainerStyle 조정함
- flex: 1;
  display: flex;
  flex-direction: column;
  overflow-y: auto;
  padding-bottom: 2rem;

dyeongg committed:
feat: Api 호출해서 무한 스크롤 (#148)

Api 호출하여 무한 스크롤

clara-shin committed:
fix: 미니플레이어 떠 있을때 가정하여 링크추가버튼 위치 수정

clara-shin committed:
design: 자잘한 스타일 수정

unanbb committed:
feat: 댓글 삭제 시 commentCount 감소 및 현재 댓글 수 연동 (#115)

- 댓글 삭제 시 commentCount 감소 구현
- 현재 댓글 수 commentCount와 연동
- 불필요한 코드 삭제
- commentCount 일치하지 않는 데이터 수정

unanbb committed:
chore: 불필요한 코드 삭제 추가 커밋 (#115)

clara-shin committed:
fix: 로그인한 사용자의 플레이리스트 목록만 순서변경 가능하도록 수정

- 로그인한 사용자 아이디로 해당 플레이리스트 데이터의 userId와 비교
- 같은 사용자 아이디일 경우에만 순서변경 가능하도록 수정
- 로그인한 유저의 아이디와 해당 플레이리스트의 userId가 다를 경우 순서변경 버튼 안보이도록 수정
    - 드래그 핸들이 사라졌으므로 목록의 좌측 여백을 1rem으로 설정

dyeongg committed:
feat: 최근 업데이트된 플리 순으로 정렬 (#148)

최근 업데이트 순서로 플레이리스트 정렬

dyeongg committed:
feat: 불필요한 코드 삭제, 다른 탭 이동후 돌아올때 데이터가 중복되는 버그 해결 (#148)

- 불필요한 코드 삭제
- 다른 탭 이동후 돌아올때 데이터가 중복되는 버그 해결

dyeongg committed:
Merge pull request #164 from Dev-FE-1/feat/search-scroll-#163

[feat] 탐색페이지 카테고리 가로 스크롤 구현

miniseung committed:
Merge pull request #169 from Dev-FE-1/feat/newPlaylist-refetch-#151

[Fix] PlaylistPage와 VideoModal 컴포넌트의 버그 픽스

miniseung committed:
feat: playlistpage api mutation 함수 (#158)

miniseung committed:
feat: 타입지정 (#158)

miniseung committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into feat/playlist-mutation-#158

miniseung committed:
feat: 드래그앤드랍 낙관적업데이트 (#158)

miniseung committed:
chore: css 수정 (#158)

miniseung committed:
Merge pull request #170 from Dev-FE-1/feat/playlist-mutation-#158

[Feat] 플레이리스트 메인 mutation

unanbb committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into feat/commentPage-#115

clara-shin committed:
fix: 플레이리스트의 업로더를 클릭하면 해당 업로더의 마이페이지가 아닌 내 마이페이지로 이동하는 버그 수정 (#171)

- useUserData함수 내의 세션으로부터 아이디를 가져오는 로직을 분리하여, 파라미터값을 외부로 부터 받아오도록 변경
- 마이페이지에서(사용하는 페이지에서) 세션으로부터 아이디를 가져옴
- 마이페이지에 외부로 받아오는 userId를 useParams로 받아오고,
- useUserData함수의 파라미터로 넣어줌. 그런데 만약 값이 없다면 세션의 userId로 대체함

clara-shin committed:
Merge branch 'develop' into feat/fork-toggle-button-#132

clara-shin committed:
Merge pull request #159 from Dev-FE-1/feat/fork-toggle-button-#132

[Feat] 구독 토글 버튼에 API 연결해 클릭마다 요청 보내는 기능 구현

clara-shin committed:
Merge pull request #168 from Dev-FE-1/feat/commentPage-#115

[Feat] 내 댓글 삭제하기 추가 작업

clara-shin committed:
Merge pull request #172 from Dev-FE-1/fix/routing-mypage-#171

[Fix] 플레이리스트의 업로더를 클릭하면 해당 업로더의 마이페이지가 아닌 내 마이페이지로 이동하는 버그 수정

itsnowkwon committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into feat/infinite-scroll-#148

clara-shin committed:
Merge pull request #167 from Dev-FE-1/feat/infinite-scroll-#148

[Feat] 홈페이지 최신 플리 무한스크롤, 업데이트 순서로 정렬

clara-shin committed:
style: mutation 함수명 변경

함수명에 mutation이라는 단어가 들어가면 너무 길어질 것 같아서 mutation을 제거하고 함수명만 작성

clara-shin committed:
style: 경로수정 및 불필요한 코드 삭제

itsnowkwon committed:
fix: 구독 버튼 클릭 시, 데이터베이스에 카운트되지 않는 버그 수정 (#160)
- 플레이리스트의 `forkCount` 값이 버튼 클릭에 따라 +1, -1될 수 있게 수정
- `userPlaylist` 컬렉션에 플레이리스트와의 관계를 정의해 포크한 플레이리스트 파악

itsnowkwon committed:
fix: MyPage의 현재 사용자인지 다른 사용자인지에 따른 동적 렌더링 버그 수정 (#160)

- 기존의 경우, 프로필만 동적 렌더링이 되고 플레이리스트 부분은 항상 현재 사용자의 플레이리스트가 렌더링되는 버그
- URL의 `userId`와 세션의 `userId`를 대조하여 동적 렌더링
- 공개/비공개 여부에 따른 플레이리스트 수 동적 렌더링

clara-shin committed:
feat: VideoModal에서 각 동영상 삭제 기능 추가

- VideoModal에서 각 동영상 삭제 기능 추가
- 동영상 삭제 시 해당 동영상이 곧바로 삭제되며 리렌더링: 남은 동영상만 보여짐
- 동영상 삭제 성공 시 토스트 메시지로 삭제 성공 알림
- 동영상 삭제 시 해당 동영상이 재생 중이었다면 다음 동영상으로 자동 재생

itsnowkwon committed:
fix: api > endpoints 중복 함수 삭제 (#160)

- `playlist.ts`에 중복 함수 삭제
- 관심사에 따라 조회 / 수정, 삭제, 추가로 파일 분리

clara-shin committed:
feat: 비디오 모달에서 영상 순서변경하면 상위에 보이는 영상 순서도 변경

드래그앤드롭으로 영상 순서 변경 시, 변경된 순서대로 인덱스값이 변경되어 보여짐

clara-shin committed:
Merge pull request #174 from Dev-FE-1/fix/playlistpage-#174

[Feat] 비디오모달에서 동영상 삭제 기능 추가 및 업데이트된 동영상 순서대로 인덱스 보여지는 기능 추가

itsnowkwon committed:
fix: 동영상 등록 후 다시 등록할 때 기존 URL 남아있는 버그 해결 (#160)

itsnowkwon committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into refactor/playlists-customhook-#160

itsnowkwon committed:
Merge pull request #175 from Dev-FE-1/refactor/playlists-customhook-#160

[Refactor] 플레이리스트 구독과 관련된 버그 해결

clara-shin committed:
design: 디자인 수정 (#176)

clara-shin committed:
Merge pull request #177 from Dev-FE-1/design/total-editing-#176

[design] 디자인 수정

itsnowkwon committed:
fix: 플레이리스트 등록 시, 이미지 등록을 하지 않을 경우를 대비한 디퐅르 이미지 추가 (#160)

itsnowkwon committed:
fix: 메인페이지 비공개 플레이리스트 렌더링되는 오류 수정 (#160)

itsnowkwon committed:
rename: `router` 역할을 하는 `Router.tsx` 파일 `App.tsx` 파일에서 분리 (#160)

itsnowkwon committed:
style: 불필요한 주석 삭제 (#160)

itsnowkwon committed:
Merge pull request #178 from Dev-FE-1/refactor/playlists-customhook-#160

[Refactor] 이미지가 없을 경우 디폴트 이미지 추가, 비공개 플레이리스트 렌더링되는 오류 수정

unanbb committed:
feat: 댓글 필터링 기능 추가 (#115)

댓글 필터링 기능 추가 및 관련 css 수정

unanbb committed:
fix: refetch시 playlistId 받아오지 못하는 오류 해결 (#115)

props로 playlistId를 받아오도록 수정

itsnowkwon committed:
refactor: 구독페이지 플레이리스트 없을 경우, 메세지 출력 (#160)

itsnowkwon committed:
rename: `contants`의 `normal.ts`파일 앞 공백 삭제 (#160)

itsnowkwon committed:
Merge branch 'feat/commentPage-#115' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into feat/commentPage-#115

itsnowkwon committed:
Merge pull request #179 from Dev-FE-1/refactor/playlists-customhook-#160

[Refactor] 구독페이지 플레이리스트 없을 경우, 메세지 출력

itsnowkwon committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into feat/commentPage-#115

itsnowkwon committed:
Merge pull request #180 from Dev-FE-1/feat/commentPage-#115

[Feat] 댓글 최신순 필터링

itsnowkwon committed:
fix: 검색페이지 플레이리스트 5개만 나오는 버그 수정 (#160)

- 파이어베이스 `limit` 기본값이 5로 설정되어있었음

itsnowkwon committed:
Merge pull request #181 from Dev-FE-1/refactor/playlists-customhook-#160

[Fix] 검색페이지 플레이리스트 5개만 나오는 버그 수정

dyeongg committed:
fix: 홈페이지 최신 플리에서 1~5번 데이터만 뜨는 버그 해결 (#183)

dyeongg committed:
Merge pull request #184 from Dev-FE-1/scroll-bugfix-#183

[fix] 홈페이지 최신 플리에서 1~5번 데이터만 뜨는 버그 해결

itsnowkwon committed:
fix: 인기 플레이리스트 좋아요 0인 플레이리스트는 보이지 않게 수정 (#160)

itsnowkwon committed:
Merge pull request #186 from Dev-FE-1/refactor/playlists-customhook-#160

[Fix] 인기 플레이리스트 좋아요 0인 플레이리스트는 보이지 않게 수정

unanbb committed:
fix: 댓글 필터링 오류 수정 (#115)

- originalComments 변수를 추가해서 sort 전 마지막 comments를 저장하도록
  하였음

unanbb committed:
Merge pull request #190 from Dev-FE-1/feat/commentPage-#115

[Fix] 댓글 필터링 오류 수정

clara-shin committed:
fix: 비디오 모달에서 비디오 삭제 시 바로 삭제되고 재 랜더링 되는 기능

clara-shin committed:
Merge pull request #192 from Dev-FE-1/fix/videomodal

fix: 비디오 모달에서 비디오 삭제 시 바로 삭제되고 재 랜더링 되는 기능

clara-shin committed:
fix: 비디오 모달에서 비디오 영상 순서 변경 시 변경된 인덱스로 상위에 보여지도록 수정

clara-shin committed:
Merge pull request #193 from Dev-FE-1/fix/videomodal

[Fix] 비디오 모달에서 비디오 영상 순서 변경 시 변경된 인덱스로 상위에 보여지도록 수정

miniseung committed:
feat:로그인 엔터키 적용 및 auth 적용 (#185)

miniseung committed:
Merge pull request #194 from Dev-FE-1/feat/login-auth-#185

[fix] 로그인페이지 엔터키 적용 및 로그인 로그아웃 auth 인증 사용

miniseung committed:
feat: 라우팅 로그인x,잘못된url->로그인 로그인0,잘못된된url->404 (#185)

miniseung committed:
chore:변수명 간소화 (#185)

miniseung committed:
feat:유효하지않은 404페이지 연결 (#185)

miniseung committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into feat/login-auth-#185

miniseung committed:
Merge pull request #195 from Dev-FE-1/feat/login-auth-#185

[fix] 로그인여부에 따라 로그인페이지로 이동 및 404페이지 이동

miniseung committed:
fix: 플리수정시 썸네일이미지 로드 안되는현상 수정 (#196)

miniseung committed:
Merge pull request #197 from Dev-FE-1/feat/playlistedit-bug-#196

[fix] 플레이리스트 수정시 기존 썸네일 이미지 안나오는 현상

itsnowkwon committed:
fix: TanStack Query stale time 2000ms로 수정 (#160)

itsnowkwon committed:
Merge pull request #198 from Dev-FE-1/refactor/playlists-customhook-#160

[Fix] TanStack Query stale time 2000ms로 수정

clara-shin committed:
design: 토스트 보이는시간 수정

2초 -> 1.5초로 수정

clara-shin committed:
design: 디폴트 이미지 추가 및 변경

플레이리스트 디폴트 썸네일
유저 디폴트 이미지

clara-shin committed:
docs: README.md 1차 업로드

- 목표 추가 및 구체화 필요
- 나머지 퇴고 및 필요한 내용 변경 및 수정 요망

clara-shin committed:
Merge pull request #199 from Dev-FE-1/design/total-editing-#176

[Design] total editing & 리드미 1차 업로드

clara-shin committed:
docs: 리드미 대문이미지 추가

clara-shin committed:
Merge pull request #200 from Dev-FE-1/design/total-editing-#176

docs: 리드미 대문이미지 추가

clara-shin committed:
docs: 리드미 사용자페이지 이미지 업로드

clara-shin committed:
Merge pull request #202 from Dev-FE-1/design/total-editing-#176

docs: 리드미 사용자페이지 이미지 업로드

dyeongg committed:
feat: 구독 페이지 없을 때 표시 (#201)

구독한 목록이 없을때 표시하는 데이터

dyeongg committed:
Merge pull request #203 from Dev-FE-1/feat/subscription-page-#201

[feat] 구독 페이지 없을 때 표시

clara-shin committed:
docs: README.md 내용수정

clara-shin committed:
Merge pull request #204 from Dev-FE-1/design/total-editing-#176

docs: README.md 내용수정중

dyeongg committed:
docs: Update README (#189)

clara-shin committed:
design: 전체 스타일수정

dyeongg committed:
Merge pull request #205 from Dev-FE-1/docs/Readme-#189

[docs] Update README (#189)

clara-shin committed:
Merge pull request #206 from Dev-FE-1/design/total-editing-#176

design: 전체 스타일수정

itsnowkwon committed:
fix: 댓글 추가 시 플레이리스트 페이지에서 생기는 라우팅 버그 해결 (#182)

- 댓글 추가 후 뒤로가기로 플레이리스트로 라우팅
- 플레이리스트에서 뒤로가기하면 그 전의 페이지가 아닌 댓글 페이지로 이동하는 버그
- 플레이리스트에 진입 시 이전 페이지를 전역 상태로 갖고 있다가 플레이리스트에서 뒤로가기 시 이 URL로 라우팅

itsnowkwon committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into fix/navigate-bug-#182

itsnowkwon committed:
Merge pull request #208 from Dev-FE-1/fix/navigate-bug-#182

[Fix] 댓글 추가 시 플레이리스트 페이지에서 생기는 라우팅 버그 해결

itsnowkwon committed:
fix: 업데이트 일자가 0일 전으로 나오는 버그 수정 (#182)

- 현재 1시간보다 클 경우에만 시간 단위가 적용이었기에 1시간 미만의 경우 0일 전으로 나왔음
- 1 이상으로 단위를 수정

itsnowkwon committed:
fix: 메인페이지 내 관심사와 비슷한 플레이리스트에서 나의 플레이리스트도 나오는 버그 수정 (#182)

- 필터링된 플레이리스트에서 현재 사용자의 `userId`와 동일한 플레이리스트는 제외

itsnowkwon committed:
Merge pull request #210 from Dev-FE-1/fix/navigate-bug-#182

[Fix] 메인페이지 내 관심사와 비슷한 플레이리스트에서 나의 플레이리스트도 나오는 버그 수정

dyeongg committed:
feat: 로그인 후 플리 추가하는 테스트코드 작성 (#209)

playwright로 로그인 후 플레이리스트를 추가하는 로직 테스트코드를 작성하였습니다

dyeongg committed:
Merge pull request #211 from Dev-FE-1/feat/playwright-test-#209

[feat] 로그인 후 플리 추가하는 테스트코드 작성

clara-shin committed:
docs: README 파일구조 추가

clara-shin committed:
Merge pull request #212 from Dev-FE-1/design/total-editing-#176

[docs] README 파일구조 추가

unanbb committed:
docs: 리드미 수정 (#213)

역할 및 오탈자 수정

clara-shin committed:
Merge pull request #214 from Dev-FE-1/fix/readme

[Docs] 리드미 수정

dyeongg committed:
Merge pull request #215 from Dev-FE-1/develop

디벨롭에서 메인으로 합병

clara-shin committed:
docs: 리드미 최종 수정

itsnowkwon committed:
fix: 좋아요 1000이 넘을 때 NaN으로 나오는 버그 해결 (#217)

- 1000이 넘을 경우 `formatNumberToK` 함수로 0.0K 형식으로 변환
- 좋아요수를 보여주는 컴포넌트의 경우 Like 타입이 number이지만, 0.0K형식은 string
- 컴포넌트의 타입을 string으로 변경해서 수정

itsnowkwon committed:
Merge pull request #219 from Dev-FE-1/fix/formatNumber-to-k-#217

[Fix] 좋아요 1000이 넘을 때 NaN으로 나오는 버그 해결

iamidlek commented on src/components/page/playlistAdd/PlaylistForm.tsx:
```diff
@@ -0,0 +1,229 @@
+        />
+        <div css={buttonStyle}>
+          <Button type='submit' styleType={`${!titleValue.trim() ? 'disabled' : 'primary'}`}>
+            {isSubmitting ? <Spinner /> : type === 'add' ? '만들기' : '수정하기'}
```
isSubmitting 항상 false 입니다.

iamidlek commented on src/components/page/playlistAdd/PlaylistForm.tsx:
```diff
@@ -0,0 +1,229 @@
+            ref={fileInputRef}
+            onChange={handleFileChange}
+            accept='image/*'
+            style={{ display: 'none' }}
```
인라인 스타일은 추후에 유지보수가 어려워질 수 있습니다.

iamidlek commented on src/pages/PlaylistPage.tsx:
```diff
@@ -0,0 +1,395 @@
+          <IconButton Icon={isForked ? GoStarFill : GoStar} onClick={handleForkToggle} />
+        )}
+      </div>
+      <DragDropContext onDragEnd={handleDragEnd}>
```
잘 구현해 주신것 같습니다.

iamidlek commented on src/router/Router.tsx:
```diff
@@ -0,0 +1,111 @@
+              { path: PATH.PLAYLIST_EDIT, element: <PlaylistEdit /> },
+            ],
+          },
+          { path: '/section-list', element: <DetailList /> },
```
/section-list 도 변수처리하면 좋을 것 같습니다.

dyeongg replied:
완료했습니다!

iamidlek commented on src/pages/CommentList.tsx:
```diff
@@ -0,0 +1,213 @@
+  const location = useLocation();
+
+  const { toastMessage, refetchComments } = location.state || {};
+  const filterOptions = [
```
컴포넌트 외부에 선언되어도 좋을 것 같습니다.

dyeongg replied:
완료했습니다!

iamidlek commented on src/tests/SigninTest.test.ts:
```diff
@@ -0,0 +1,38 @@
+test('잘못된 자격 증명으로 로그인 실패', async ({ page }) => {
+  await page.goto('http://localhost:5173/signin');
+
+  await page.fill('input[id="username"]', 'wronguser@example.com');
```
모든 테스트에서 실행하는 beforeeach 등으로 반복 처리를 작성할 수 있습니다.

dyeongg replied:
수정했습니다!

iamidlek reviewed (commented):
고생하셨습니다.

유저가 사용하기 편하게 잘 구현해 주신 것 같습니다.
일부 데이터 처리에 있어서 반복문을 많이 도는 부분이나
같은 데이터를 여러번 부르는 등의 문제가 있는 것 같습니다.
타입 에러가 나는 곳도 있는 것 같습니다.

폴더 구조나 코드의 분리는 아주 잘 해주셨고
디자인과 애니메이션에 공을 많이 들이신 것이 느껴질 정도로
완성도가 높은 프로젝트인것 같습니다.

unanbb commented on src/pages/Subscriptions.tsx:
```diff
@@ -0,0 +1,114 @@
+const Subscriptions: React.FC = () => {
+  const [forkedPlaylists, setForkedPlaylist] = useState<PlaylistModel[]>([]);
+  // const { isToggled, toggle } = useToggleStore();
+  // const { showToast } = useToastStore();
```
제거완료 했습니다!

miniseung commented on src/pages/Signin.tsx:
```diff
@@ -0,0 +1,129 @@
+    if (isFormValid) {
+      try {
+        const userCredential = await signInWithEmailAndPassword(auth, username, password);
+        console.log('로그인 성공:', userCredential.user.email);
```
console.log 제거했습니다

unanbb committed:
refactor: 댓글 필터링 기능 리팩토링 및 데이터 상수화 작업 (#222)

- 댓글 필터링 기능 sort -> reverse로 변경 (댓글 많아질 때 처리속도 상승)
- constants폴더에 comment.ts, title.ts 생성 및 string값 상수화
- 불필요한 주석 삭제

miniseung commented on src/components/page/signin/GoogleLogin.tsx:
```diff
@@ -0,0 +1,84 @@
+        });
+      }
+
+      await setDoc(
+        userPlaylistsDocRef,
+        {
+          created: [],
+          forked: [],
+          liked: [],
+        },
+        { merge: true }
+      );
```
여기 부분이 구글로 로그인하면 구독 좋아요 생성한 목록들이 로그아웃하면 사라지는 문제가 있었는데 이 부분때문에 그런거 같습니다 
      const userPlaylistsDoc = await getDoc(userPlaylistsDocRef);

      if (!userPlaylistsDoc.exists()) {
        await setDoc(userPlaylistsDocRef, {
          created: [],
          forked: [],
          liked: [],
        });
      }
이러게 변경하여 유저가 존재하지 않을때만 실행하도록 변경하였습니다

miniseung committed:
refactor: 구글로그인시userPlaylistsDocRef초기화 조건문 (#220)

unanbb committed:
refactor: title.ts 수정 (#222)

TITLE이라는 객체로 변경하여 내부 property로 페이지 타이틀을 제어할 수
있도록 변경

dyeongg committed:
refactor: 경로 변수명으로 변경, 댓글리스트 외부선언 뽑기

- '/section-list'를 PATH.DETAIL_LIST 변수명으로 변경
- 댓글리스트 필터옵션 컴포넌트  외부선언

dyeongg committed:
refactor: 댓글리스트 재커밋 ((#227)

dyeongg committed:
refactor: 로그인 테스트 코드 beforeEach로 반복 처리 (#229)

beforeEach를 사용하여 온보딩 페이지를 반복처리함

dyeongg committed:
refactor: setTimeout 제거

setTimeout 시간딜레이 삭제

dyeongg committed:
refactor: 1번부터 컨텐츠 반복 삭제 (#231)

컨텐츠 1번부터 다시 반복하는 부분 삭제

miniseung committed:
refactor: input과 signin페이지 리펙토링 (#226)

miniseung commented on src/pages/Signin.tsx:
```diff
@@ -0,0 +1,129 @@
+  };
+
+  return (
+    <div css={container} onKeyDown={handleKeyDown}>
```
input에서 keydown을 받도록 수정했습니다

miniseung commented on src/components/common/Input.tsx:
```diff
@@ -0,0 +1,104 @@
+    }
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+  };
```
수정했습니다

miniseung commented on src/pages/Signin.tsx:
```diff
@@ -0,0 +1,129 @@
+    setUsername(newUsername);
+    setPassword(newPassword);
+  };
+
+  const handleSignIn = async (e?: React.FormEvent) => {
+    if (e) e.preventDefault();
+
+    if (isFormValid) {
```
수정했습니다

miniseung commented on src/pages/Signin.tsx:
```diff
@@ -0,0 +1,129 @@
+  const [errorMessage, setErrorMessage] = useState('');
+  const navigate = useNavigate();
+
+  const handleInputChange = (isValid: boolean, newUsername: string, newPassword: string) => {
+    setIsFormValid(isValid);
+    setUsername(newUsername);
+    setPassword(newPassword);
+  };
```
Signin에서 내려받도록 수정했습니다

miniseung commented on src/router/Router.tsx:
```diff
@@ -0,0 +1,111 @@
+    setIsOnboarding(onboardingStatus);
+
+    return () => unsubscribe();
+  }, []);
```
AuthProtectedRoute에서 로그인여부를 확인하고 로그인 하지 않은 상태로 다른 페이지에 접근을 하지 못하게 하기 위해서입니다

red-dev-Mark commented on src/hooks/useSortedPlaylists.ts:
```diff
@@ -0,0 +1,64 @@
+  const errorForAllForkedPlaylist = forkedPlaylistsQueries.find((query) => query.error);
+  const allForkedPlaylists = forkedPlaylistsQueries
+    .flatMap((query) => query.data || [])
+    .filter((playlist) => playlist.userId !== userId)
+    .filter((playlist) => playlist.isPublic === true);
```
수정했습니다.

red-dev-Mark commented on src/components/page/home/CrossScrollingList.tsx:
```diff
@@ -0,0 +1,137 @@
+  ALL: '전체보기',
+};
+
+const CrossScrollingList: React.FC<CrossScrollingListProps> = ({ title, playlists }) => {
```
수정했습니다.

red-dev-Mark commented on src/components/page/home/CrossScrollingList.tsx:
```diff
@@ -0,0 +1,137 @@
+              type='main1'
+              thumURL={playlist.thumbnailUrl}
+              title={playlist.title}
+              likes={+formatNumberToK(playlist.likeCount)}
```
수정했습니다.

red-dev-Mark commented on src/main.tsx:
```diff
@@ -1,9 +1,15 @@
 import { StrictMode } from 'react';
```
수정했습니다.

red-dev-Mark commented on src/components/common/ThumbNailBox.tsx:
```diff
@@ -0,0 +1,303 @@
+    }
+  };
+
+  return renderContent();
```
수정했습니다.

red-dev-Mark commented on src/components/page/search/FilteredPlaylists.tsx:
```diff
@@ -0,0 +1,55 @@
+  const playlists = displayedPlaylists?.filter((playlist) => playlist.isPublic === true);
+  return (
+    <div>
+      {playlists && playlists.length > 0 ? (
```
수정했습니다.

itsnowkwon committed:
refactor: 메인페이지의 불필요한 주석 삭제 및 로직 단순화

- 불필요한 주석 삭제
- 삼항연산자를 통한 로직 단순화

clara-shin committed:
Merge pull request #224 from Dev-FE-1/refactor/comments-#222

[Refactor] 댓글 필터링 기능 리팩토링 및 데이터 상수화 작업

clara-shin committed:
Merge pull request #225 from Dev-FE-1/refactor/singin-#220

[Refactor] 구글 로그인시 userPlaylists 초기화 문제 해결

clara-shin committed:
Merge pull request #228 from Dev-FE-1/refactor/commentList-#227

[refactor] section-list 경로 변수명으로 변경

clara-shin committed:
Merge pull request #230 from Dev-FE-1/refactor/testcode-#229

[refactor] 로그인 테스트 코드 beforeEach로 반복 처리

clara-shin committed:
Merge pull request #232 from Dev-FE-1/refactor/scroll-#231

[Refactor] 무한스크롤 코드 내 시간딜레이, 컨텐츠 반복 삭제

clara-shin committed:
Merge pull request #236 from Dev-FE-1/refactor/signin-inputform-#226

[Refactor] input과 signin페이지 리펙토링

dyeongg commented on src/hooks/useDragToScroll.ts:
```diff
@@ -0,0 +1,40 @@
+  const onMouseDown = (e: React.MouseEvent) => {
+    setIsDragging(true);
+    setDragDistance(0);
+    setStartX(e.pageX - e.currentTarget.offsetLeft);
```
수정했습니다

dyeongg commented on src/components/page/search/CategoryButtons.tsx:
```diff
@@ -0,0 +1,98 @@
+import { useRef, useState } from 'react';
```
삭제했습니다

dyeongg commented on src/components/page/search/CategoryButtons.tsx:
```diff
@@ -0,0 +1,98 @@
+
+interface CategoryButtonsProps {
+  selectedCategory: string;
+  handleButtonClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
```
eslint 설정 수정했습니다!

dyeongg commented on src/components/page/playlist/PlaylistBox.tsx:
```diff
@@ -0,0 +1,212 @@
+    };
+
+    fetchUserData();
+  }, []);
```
수정했습니다

dyeongg commented on src/pages/Settings.tsx:
```diff
@@ -0,0 +1,98 @@
+          type='alertConfirm'
+          isOpen={isModalOpen}
+          onClose={closeModal}
+          onConfirm={confirmSignOut}
```
수정했습니다

dyeongg committed:
refactor: 타입에러, lint에러, 사용되지않은 선언 제거

- lint no-unused-vars': 'error' 설정변경
- Dialog 필수 props 변경
- categoryButtons 사용되지 않는 값 제거

clara-shin committed:
fix: 스피너 위치 수정 및 관련 중복코드 삭제 (#239)

clara-shin committed:
Merge pull request #238 from Dev-FE-1/refactor/type-error-#237

[refactor] 타입에러, lint에러, 사용되지않은 선언 제거

unanbb committed:
feat: 댓글 프로필 이미지 라우팅 (#223)

댓글 프로필 이미지 클릭 시 해당 사용자의 페이지로 이동하는 handle함수
구현

unanbb committed:
feat: 내 댓글 프로필 라우팅 처리 해제 (#223)

다른 사용자의 프로필 이미지만 개인 프로필 페이지로 이동하도록 변경

unanbb committed:
chore: userSession 부르는 위치 변경 (#223)

userSession을 불러오는 위치를 상위스코프로 변경

miniseung committed:
refactor: router authprotected 함수 의존성 추가, 세션값없을시 로그아웃 (#221)

miniseung committed:
chore: 불필요한코드제거 (#221)

miniseung committed:
fix:동영상삭제모달후 플리삭제모달안뜨는 버그 픽스 (#248)

dyeongg committed:
Merge pull request #240 from Dev-FE-1/fix/spinner

[Fix] 스피너 위치 수정 및 관련 중복코드 삭제

dyeongg committed:
refactor: 로그인테스트코드 온보딩처리 수정 (#229)

waitfortimeout으로 세션에 값이 들어가는 시간 대기

clara-shin committed:
feat: 비디오 삭제 시 새로 인덱스 업데이트 기능 추가 (#242)

- 비디오 삭제 시 새로운 인덱스 업데이트 기능 추가
    - 재생 중인 비디오 삭제 시, 다음 비디오로 자동 재생되도록 기능 추가
        - 이때 삭제된 비디오의 인덱스를 기준으로 다음 비디오의 인덱스를 업데이트
        - 전체 비디오 목록에서 삭제된 비디오를 제외하고 총 갯수를 업데이트 함
        - 삭제되고 남은 비디오의 인덱스는 그대로 유지
        - 삭제된 비디오가 마지막 비디오인 경우,
            - 그 다음 비디오가 없으므로 재생 중인 비디오를 초기화
            - 영상이 멈춘 상태로 유지
    - 재생 중인 비디오가 아니라면, 재생중인 비디오를 초기화하지 않음
        - 삭제된 비디오의 인덱스를 기준으로 다음 비디오의 인덱스를 업데이트
        - 전체 비디오 목록에서 삭제된 비디오를 제외하고 총 갯수를 업데이트 함
    - 더이상 삭제할 비디오가 없는 경우, 모달을 닫음

clara-shin committed:
Merge pull request #244 from Dev-FE-1/feat/comments-#223

[Feat] 댓글 프로필 클릭 시 개인 페이지로 라우팅 설정

clara-shin committed:
Merge pull request #249 from Dev-FE-1/feat/playlist-modal-#248

[Fix] 플레이리스트 메인 모달 오류

clara-shin committed:
fix: 안쓰는 변수 삭제 및 eslint 수정

95번째 줄 안쓰는 변수 관련 warn으로 재변경
비디오 모달에서 안쓰는 변수 최대한 삭제

clara-shin committed:
Merge branch 'develop' into refactor/main

clara-shin committed:
Merge pull request #246 from Dev-FE-1/refactor/main

[Refactor] formatNumberToK 타입 에러로 생기는 버그 수정

clara-shin committed:
Merge pull request #247 from Dev-FE-1/refactor/router-#221

[Refactor] 라우터 어스프로텍트 의존성 추가

clara-shin committed:
Merge pull request #250 from Dev-FE-1/refactor/testcode-#229

[refactor] 로그인 테스트코드 온보딩 처리 수정

clara-shin committed:
Merge pull request #251 from Dev-FE-1/feat/deleteVideoUpdate-#242

feat: 비디오 삭제 시 새로 인덱스 업데이트 기능 추가 (#242)

clara-shin committed:
Merge pull request #252 from Dev-FE-1/fix/videoModalPage

[Fix] 안쓰는 변수 삭제 및 eslint 수정

unanbb committed:
refactor: CommentBox.tsx 코드 개선 (#223)

- CommentBox props 간소화
- 함수 개선 및 코드 간소화
- 댓글 삭제 에러 처리 토스트 추가
- 중복 코드 변수화

dyeongg committed:
feat: 가로스크롤 버튼 추가 (#253)

가로스크롤 힌트, 가로스크롤 버튼으로 3개씩 넘기기

dyeongg committed:
refactor: useScrollWithArrows 수정 (#253)

clara-shin committed:
Merge pull request #254 from Dev-FE-1/feat/comments-#223

[Refactor] CommentBox.tsx 코드 개선

clara-shin committed:
Merge pull request #255 from Dev-FE-1/feat/horizontal-scroll-button-#253

[feat] 가로스크롤 버튼 추가

itsnowkwon committed:
feat: 구독 페이지의 구독 버튼이 동시에 적용되는 버그 해결

unanbb committed:
feat: 댓글 삭제 시 모달창 생성 (#245)

- 댓글 삭제 시 모달창 생성
- modal 상태 지역으로 관리
- 불필요한 props 삭제
- Dialog.tsx youtubeUrl관련 코드 수정

itsnowkwon committed:
Merge branch 'develop' of https://github.com/Dev-FE-1/ToyProject_3_Team2 into refactor/main

fix: Merge 후 머지 페이지에서 생기는 버그 해결
- 가로스크롤 전체보기 보이지 않는 버그 해결
- 버튼 안의 버튼을 배치해 생기는 버그 해결
- 잘못된 스피너 호출로 생기는 버그 해결
- 각 리스트 호출 커스텀훅 리팩토링

dyeongg committed:
Merge pull request #256 from Dev-FE-1/refactor/main

[Refactor] Merge 후 머지 페이지에서 생기는 버그 해결

dyeongg committed:
fix: 메인페이지에서 출력되는 오류 해결 (#259)

- onMouseUpOrLeave에 대한 오류 삭제
- 초기 렌더시 인기 플리에 스크롤버튼이 안뜨는 문제 setsetTimeout으로 가시성 업데이트

clara-shin committed:
fix: 자세히보기 버튼 아이콘 위치 수정 (#261)

홈 화면의 자세히보기 버튼 아이콘 위치를 수정
마이페이지의 자세히보기 버튼 아이콘 위치를 수정

clara-shin committed:
Merge branch 'develop' into feat/deleteModal-#245

clara-shin committed:
Merge pull request #258 from Dev-FE-1/feat/deleteModal-#245

feat: 댓글 삭제 시 모달창 생성 및 필터링 버그 수정(#245)

clara-shin committed:
Merge pull request #260 from Dev-FE-1/fix/mainpage-error-#259

[fix] 메인페이지에서 출력되는 오류 해결

dyeongg committed:
Merge pull request #262 from Dev-FE-1/fix/iconbutton-detail-#261

fix: 자세히보기 버튼 아이콘 위치 수정 (#261)

miniseung committed:
fix:온보딩보다로그인이 먼저 랜더링되는 이슈 (#257)

clara-shin committed:
fix: react-beautiful-dnd과 @types/react-beautiful-dnd를 devDependencies로 이동

clara-shin commented on package.json:
```diff
@@ -6,24 +6,62 @@
+    "@radix-ui/react-switch": "^1.1.0",
+    "@tanstack/react-query": "^5.52.1",
+    "@tanstack/react-query-devtools": "^5.52.1",
+    "@types/react-beautiful-dnd": "^13.1.8",
```
devDependencies로 변경완료

clara-shin commented on tsconfig.node.json:
```diff
@@ -9,6 +9,8 @@
     "moduleResolution": "bundler",
     "allowSyntheticDefaultImports": true, // 기본 내보내기가 없는 모듈에서도 기본 import를 허용
     "isolatedModules": true,
+    "jsx": "react-jsx",
+    "jsxImportSource": "@emotion/react",
```
저 설정을 해주지 않으면 @emotion/react를 사용할 때마다 각 페이지에 관련 주석을 넣어주어야 해서 그냥 두었습니다.

clara-shin commented on src/layouts/RootLayout.tsx:
```diff
@@ -0,0 +1,49 @@
+
+const RootLayout = () => {
+  const location = useLocation();
+  const isOpen = useMiniPlayerStore((state) => state.isOpen);
+  const videoId = useMiniPlayerStore((state) => state.videoId);
+  const playlist = useMiniPlayerStore((state) => state.playlist);
+  const userId = useMiniPlayerStore((state) => state.userId);
```
한꺼번에 함수만 가져오면 브레이스를 쓸 수 있지만, 모든 속성을 브레이스에 가져오면 안된다고 하여서.. 그냥 두었습니다.

dyeongg committed:
Merge pull request #263 from Dev-FE-1/fix/router-#257

[Fix] 크롬 첫 진입시 온보딩페이지보다 로그인페이지가 먼저 랜더링되는 버그

dyeongg committed:
Merge pull request #264 from Dev-FE-1/fix/settings

[Fix] settings

clara-shin committed:
Merge pull request #265 from Dev-FE-1/develop

1차 리팩토링 후, Develop에서 Main으로 2차 합병
